packages/compiler/react.ts

```diff
@@ -0,0 +1,120 @@
+import * as t from '@babel/types';
+import type { NodePath } from '@babel/core';
+
+export const optimizeReact = (path: NodePath<t.CallExpression>) => {
```

Comment:
Could we add an example of input -> output code transform? I'm not sure to understand exactly how this works

Reply:
Yep, do note that this is stil very much a work in progress though:

Input:
```js
const Component = () => {
  const [state, setState] = useState(0);
  return (
    <button
      onClick={() => {
        setState(state + 1);
      }}
    >
      {state}
    </button>
  );
};

const CBlock = block(Component);
```

Output:
```js
const CBlock = () => {
  const [state, setState] = useState(0);
  return _block$({
    _prop0$: () => {
      setState(state + 1);
    },
    state
  });
};
function _block$({
  _prop0$,
  state
}) {
  return /* @__PURE__ */ React.createElement("button", { onClick: _prop0$ }, state);
}
const Component = block(_block$);
```

Reply:
Currently only supports identifiers and arrow expressions, hope to extend to all identifiers + expressions with identifiers

Reply:
UPDATED:

Input:

```jsx
const Component = () => {
  const [state, setState] = useState(0);
  const someObject = { test: 1, test2: 2 };
  const attrs = { title: 'test' };
  const handleClick = () => {
    setState(state + 1);
  };

  return (
    <button
      onClick={handleClick}
      onMouseDown={() => console.log(someObject)}
      {...attrs}
    >
      {state}
      {someObject.test + 1}
      {someObject.test2}
    </button>
  );
};

const CBlock = block(Component);
```

Output:

```jsx
const CBlock = () => {
  const [state, setState] = useState(0);
  const someObject = {
    test: 1,
    test2: 2
  };
  const attrs = {
    title: "test"
  };
  const handleClick = () => {
    setState(state + 1);
  };
  return _block$({
    handleClick,
    __1: () => console.log(someObject),
    attrs,
    state,
    __4: someObject.test + 1,
    __5: someObject.test2
  });
};
function _block$({
  handleClick,
  __1,
  attrs,
  state,
  __4,
  __5
}) {
  return /* @__PURE__ */ React.createElement("button", { onClick: handleClick, onMouseDown: __1, ...attrs }, state, __4, __5);
}
const Component = block(_block$);
```